extensions/notebook/src/dialog/managePackages/installedPackagesTab.ts

```diff
@@ -67,20 +67,42 @@ export class InstalledPackagesTab {
 			this.installedPackagesTable = view.modelBuilder.table()
 				.withProperties({
 					columns: [
-						localize('managePackages.pkgNameColumn', "Name"),
-						localize('managePackages.newPkgVersionColumn', "Version")
+						{
+							value: localize('managePackages.pkgNameColumn', "Name"),
+							type: azdata.ColumnType.text
+						},
+						{
+							value: localize('managePackages.newPkgVersionColumn', "Version"),
+							type: azdata.ColumnType.text
+						},
+						{
+							value: localize('managePackages.deleteColumn', "Delete"),
+							type: azdata.ColumnType.button,
+							options: {
+								icon: {
+									dark: this.dialog.extensionContext.asAbsolutePath('resources/dark/delete_inverse.svg'),
+									light: this.dialog.extensionContext.asAbsolutePath('resources/light/delete.svg')
+								}
+							}
+						}
 					],
 					data: [[]],
 					height: '600px',
 					width: '400px'
 				}).component();
+			this.installedPackagesTable.onCellAction(async (rowState) => {
```

Comment:
all of the event handlers should be disposed properly

Reply:
none of the events in this dialog is disposed, is it something we should enforce for extensions?

---
In reply to: [446430280](https://github.com/microsoft/azuredatastudio/pull/10918#discussion_r446430280) [](ancestors = 446430280)

Reply:
we should, at least the ones you are adding since there could be a lot of rows.